pkg/cloudprovider/openstack/openstack.go

```diff
@@ -443,6 +443,48 @@ func (os *OpenStack) TCPLoadBalancer() (cloudprovider.TCPLoadBalancer, bool) {
 	return &LoadBalancer{network, compute, os.lbOpts}, true
 }
 
+func isNotFound(err error) bool {
+	e, ok := err.(*gophercloud.UnexpectedResponseCodeError)
+	return ok && e.Actual == http.StatusNotFound
+}
+
+func getPoolByName(client *gophercloud.ServiceClient, name string) (*pools.Pool, error) {
+	opts := pools.ListOpts{
+		Name: name,
+	}
+	pager := pools.List(client, opts)
+
+	poolList := make([]pools.Pool, 0, 1)
+
+	err := pager.EachPage(func(page pagination.Page) (bool, error) {
+		v, err := pools.ExtractPools(page)
+		if err != nil {
+			return false, err
+		}
+		poolList = append(poolList, v...)
+		if len(poolList) > 1 {
+			return false, ErrMultipleResults
+		}
+		return true, nil
+	})
+	if err != nil {
+		if e, ok := err.(*gophercloud.UnexpectedResponseCodeError); ok {
```

Comment:
Oh indeed!  I distinctly recall making that change already, so it must have been lost in some git remangling.  Retyped :/